pkg/commonutils/common_utils.go

```diff
@@ -18,3 +18,16 @@ func MeasureOperation(operation string, log logrus.FieldLogger, metricsApi metri
 		}
 	}
 }
+
+func MeasureOperationWithThresholdAndId(operation string, log logrus.FieldLogger, metricsApi metrics.API, thresholdInSec float64, objectId string) func() {
```

Comment:
If you wish to log and report metrics only in case the duration exceeded the threshold the function should be renamed, e.g.  
`MeasureOperationAndIDIfExceededThreshold`